Feature/pro templates

ultramarine64 opened the pull request:
На момент 13.07 можно потихоньку смотреть, но скорее всего прилетят какие-нибудь фиксы позже.

ultramarine64 committed:
Изменить логику работы с рефами для изображений.

ultramarine64 committed:
Реализовать третий про-шаблон.

ultramarine64 committed:
Реализовать четвертый про-шаблон.

ultramarine64 committed:
Добавить возможность задания цвета бэкграунда.

ultramarine64 committed:
Реализовать пятый про-шаблон.

ultramarine64 committed:
Добавить правки после ребейза на свежий main для возвращения работоспобности.

ultramarine64 committed:
Добавить правки для улучшения ситуации с экспортом подписи.

- Заменить v-show в таблицах на v-if
- Добавить, где это возможно, v-if для минимизации пустых блоков в таблице

ultramarine64 committed:
Улучшить читаемость инлайн-стилей.

ultramarine64 committed:
Прочие мелкие правки для улучшения читаемости.

ultramarine64 commented on sections/gallery-page.liquid:
```diff
@@ -1470,6 +2043,10 @@
         linkEl.click();
+      },
+
+      getFontSizeBasedValue( value ) {
```
Я помню, что насчет этого названия мы так и не определились, но я не смог придумать более подходящего. По сути мы ведь и получаем какое-то значение, основанное на текущем размере шрифта.

filya63 replied:
что если не value а em? Это же относительно того шрифта, что сейчас установлено

ultramarine64 replied:
``` getFontSizeBasedEm ``` ?
Не понял логику названия метода. У нас геттер, который возвращает какое-то значение (```value```), которое базируется (```based```) на текущем значении параметра размера шрифта (```fontSize```).
Это полученное значение мы не обязаны использовать в em. Сейчас, например, этот метод используется в коде для вычисления значений, которые затем задаются в пикселях.

filya63 replied:
Я имею ввиду переименовать параметр value. Так как в теле метода мы умножаем на текущий размер шрифта. Больше подходит будто em

ultramarine64 replied:
em это же вообще единица измерения, все еще не понимаю логику. Опять же, мы этот метод можем использовать как для каких-то вычислений в em, так и для вычислений в px.

filya63 replied:
но тем не менее у тебя умножения в шаблоне на 4, и так далее. Ты вызываешь эту функцию с таким аргументом. То есть если fontSize = 12 ты это умножаешь на 4? Это ли не em, во сколько раз превысить своего родителя

ultramarine64 replied:
Ладно, я твою логику понял. Как по мне, так оба варианта имеют право на жизнь, но спор особого смысла не имеет. Меняю на em.

ultramarine64 committed:
Вернуть обратно фикс бага с шейпом изображения.

ultramarine64 commented on sections/gallery-page.liquid:
```diff
@@ -1958,22 +1958,26 @@
         ref.append( imageElem );
+
+        if( isImageLink ) {
+          this.changeClassShapeForImage( imageElem, this.typeShape );
```
Как-то неловко... Упустили в прошлом ПР момент, когда этот фикс пропал.

ultramarine64 commented on sections/gallery-page.liquid:
```diff
@@ -408,6 +408,51 @@
                     </div>
                   </div>
                 </div>
+                <div class="setup__item">
```
Тут я на всякий случай уточню, откуда это появилось. При разработке пятого шаблона планировалось добавить возможно выбирать цвет бэкграунда, но только на определенном участке. Это было как-то запарно и от идеи отказались, но фича с цветом задника может пригодиться юзерам как альтернатива изображению, поэтому решили оставить.

ultramarine64 replied:
А еще много появилось копипасты с этими цветами... Работаю над упрощением всей этой логики, но уже не в этом реквесте - этот и так слишком большой.

filya63 commented on sections/gallery-page.liquid:
```diff
@@ -412,7 +480,7 @@
           <div ref="signature" :style="stylesPreviewTemplate">
             <table
-              v-show="!isProTemplate && !isPro2Template"
+              v-if="isClassicTemplate || isHorizontalTemplate || isWideTemplate"
```
Спули из мастера изменения. Есть компьютед isFreeTemplate

ultramarine64 replied:
Готово

filya63 commented on sections/gallery-page.liquid:
```diff
@@ -1012,6 +1080,494 @@
+                        </tr>
+                        <tr v-if="hasContacts">
+                          <td :style="stylesPreviewContactContent">
+                            <a v-if="phone" :href="`tel:${phone}`" :style="stylesPreviewContactContentItemLink" >{{ phone }}</a>
```
Тут перенеси на разные строки атрибуты, и в других аналогичных местах.

ultramarine64 replied:
Забыл отписаться, готово

filya63 commented on sections/gallery-page.liquid:
```diff
@@ -1385,22 +1958,26 @@
 
       createElementWithImage( imageLink ) {
-        this.createImageElement( imageLink, this.$refs.userImage, this.stylesPreviewUserPicture );
+        this.createImageElement( imageLink, this.getImageRef(), this.stylesPreviewUserPicture, true );
```
Может не будем так хардкодить?
const isImageLink = Boolean( imageLink );
и это прокидывать вместо true

filya63 replied:
Да и вообще, будто этот параметр лишний. Можно смотреть на параметр link в методе createImageElement, нет?

ultramarine64 replied:
Как-то я тебя совсем не понял.
``` Boolean( imageLink ) ``` не имеет никакого смысла, как ты и подметил ниже - если бы информацию можно было бы почерпнуть исключительно из параметра линка, то и смысла в доп. параметре бы не было.
Касательно того, что можно смотреть на параметр ``` link ``` - вообще не понял этот момент. У нас может быть либо создание элемента как именно чисто изображения, а есть создание элемента как подписи. Подпись стилизуется по другому и на него опция шейпа не действует. Проверка на ``` link ``` нам ничего не даст, кроме того, что для элементов без линка шейп задан не будет (к слову, возможны ли такие случаи сейчас?).

filya63 replied:
Вынеси хотя бы в переменную true. А то совсем не понятно для чего она тут.

ultramarine64 replied:
Готово

filya63 commented on sections/gallery-page.liquid:
```diff
@@ -1385,22 +1958,26 @@
 
       createElementWithSignature( imageLink ) {
-        this.createImageElement( imageLink, this.getUserSignatureRef(), this.stylesPreviewUserSignature );
+        this.createImageElement( imageLink, this.getUserSignatureRef(), this.stylesPreviewUserSignature, false );
```
Аналогично

ultramarine64 committed:
Улучшить читаемость кода.

- Разнести в элементах с нескольими атрибутами сами атрибуты на отдельные строки.
- Убрать перенос единичного атрибута на отдельную строку.

ultramarine64 committed:
Исправить отображаемость выбранных шрифтов.

ultramarine64 committed:
Улучшения читаемости кода.
